### DISCUSSION TITLE
Is there a way to get the address of activation tensors?

### DISCUSSION THREAD

**Discussion by andylee-24** [microsoft/onnxruntime#17733] [Other Q&A] [answered]
Hello, I'm trying to do memory simulation with ONNX.

Is there a way to get the beginning address and size of activations (intermediate input / output buffers) of the given ONNX model
while using the defaultCPU execution providers? The attached figure depicts What I'm trying to inspect.
It would be very helpful if I know the related files, class, structures or keywords....

Thanks in advance,

TaeHan Lee.

![onnx](https://github.com/microsoft/onnxruntime/assets/45648726/3a99d21a-848b-49e1-90e2-69936ad93fa4)

**Comment by hariharans29** [comment, marked as answer]
ORT doesn't have public APIs to query these.

If you would like to study more - here is a good place to start - https://github.com/microsoft/onnxruntime/blob/14d349e29075db52eb6971569b572ebe54215596/onnxruntime/core/framework/allocation_planner.cc#L135. It has logic to deteermine if ORT can re-use a buffer/ has to allocate new buffer for activations. Please keep in mind that these are internal implementation details of ORT and they are bound to change.

**Reply by andylee-24** [reply]
Thanks a lot!
